app/src/main/java/com/example/scoutconcordia/CalendarActivity.java

```diff
@@ -42,9 +103,388 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
+        String[][] dateInfoObj = dateInfoObj(); // see method description
+        fillDayDateRows(dateInfoObj);
+        SimpleDateFormat sdfDate = new SimpleDateFormat("yyyy-MM-dd"); // date formatter to compare Date object dates
+        SimpleDateFormat sdfTime = new SimpleDateFormat("h:mm"); // time formatter
+
+        for (Event event : eventsList) {
+            Date startDate = new Date(event.getStart().getDateTime().getValue()); // start date of the event
+            Date endDate = new Date(event.getEnd().getDateTime().getValue());   // end date of the event
+
+            // even starts or ends outside school hours
+            if(getTimeHour(startDate) < 8 || getTimeHour(endDate) >= 23){
+                continue;
+            }
+
+            long length = endDate.getTime() - startDate.getTime();
+            int totalRows = (int)(length / 1000 / 60) / 15;  // (length of event)/15 to get the number of rows the event covers
+                                                             // each row is 15 mins
+            String date = sdfDate.format(startDate); // convert date to "yyyy-MM-dd" format to compare with date from
+                                                 // datInfoObj as dates in dateInfoObj are stored in "yyyy-MM-dd"
+
+            int startRow = eventStartRow(startDate, endDate);
+
+            String eventName = event.getSummary();
+            String eventLocation = event.getLocation();
+            String start = sdfTime.format(startDate);
+            String end = sdfTime.format(endDate);
+
+            if (date.equals(dateInfoObj[0][1])) {   // today
+                fillEvent(0, startRow, totalRows, eventName, eventLocation, start, end, getResources().getColor(R.color.shadeOne));
+            }
+            else if (date.equals(dateInfoObj[1][1])) { //tomorrow
+                fillEvent(1, startRow, totalRows, eventName, eventLocation, start, end, getResources().getColor(R.color.shadeTwo));
+            }
+            else if (date.equals(dateInfoObj[2][1])) { // day after
+                fillEvent(2, startRow, totalRows, eventName, eventLocation, start, end, getResources().getColor(R.color.shadeOne));
+            }
+            else if (date.equals(dateInfoObj[3][1])) {
+                fillEvent(3, startRow, totalRows, eventName, eventLocation, start, end, getResources().getColor(R.color.shadeTwo));
+            } else if (date.equals(dateInfoObj[4][1])) {
+                fillEvent(4, startRow, totalRows, eventName, eventLocation, start, end, getResources().getColor(R.color.shadeOne));
+            } else if (date.equals(dateInfoObj[5][1])) {
+                fillEvent(5, startRow, totalRows, eventName, eventLocation, start, end, getResources().getColor(R.color.shadeTwo));
+            }
+            else if (date.equals(dateInfoObj[6][1])) {
+                fillEvent(6, startRow, totalRows, eventName, eventLocation, start, end, getResources().getColor(R.color.shadeOne));
```

Comment:
Unclear what these if-else statements accomplish.
By today/tomorrow/day after, what do you mean?

Reply:
![image](https://user-images.githubusercontent.com/46825370/76707911-5b090980-66c9-11ea-90cf-0a5c53c4c9a8.png)

- it checks if the date of the event is that of today, tomorrow, the day after tomorrow, and so on. If it's today, event will be placed in the first column.

- the dateInfoObj[][] contains dates in string format for comparison. 
         - dateInfoObj[0][1] = "2020-03-15"
         - dateInfoObj[0][1] = "2020-03-16"

- fillEvent(int eventDayColumn, int startRow, int totalRows, String summary, String location, String start, String end, int colorId)  gets the Id of the text view where the event is to be placed.   Text view ids are stored in a  int[7][61] array. 
        - int[0][1]  = current day from 8:00 - 8:15
        - int[1][4] = tomorrow 8:45 - 9: 00

![image](https://user-images.githubusercontent.com/46825370/76708275-f3a08900-66cb-11ea-84ed-7a21798d376a.png)